packages/compass/src/main/application.ts

```diff
@@ -12,12 +11,21 @@ import { CompassMenu } from './menu';
 import { setupCSFLELibrary } from './setup-csfle-library';
 import { setupPreferencesAndUserModel } from './setup-preferences-and-user-model';
 import type { ParsedGlobalPreferencesResult } from 'compass-preferences-model';
+import preferences from 'compass-preferences-model';
 
-const debug = createDebug('mongodb-compass:main:application');
+import createLoggerAndTelemetry from '@mongodb-js/compass-logging';
+
+const { debug, track } = createLoggerAndTelemetry('COMPASS-MAIN');
 
 type ExitHandler = () => Promise<unknown>;
 type CompassApplicationMode = 'CLI' | 'GUI';
 
+const launchConnection = (file?: string, positionalArguments?: string) => {
```

Comment:
using `launch` as a verb here sounds a bit like this would be the function that actually launches a connection somehow, might be nice to indicate in the function name that this only computes a value based on preferences / provides a result for telemetry

Reply:
Yeah, i was going to rename it, absolutely agree.